core/templates/dev/head/pages/exploration_editor/editor_tab/StateResponses.js

```diff
@@ -259,99 +250,113 @@ oppia.controller('StateResponses', [
             var rulesService = $injector.get(rulesServiceName);
 
-            // See the training panel directive in StateEditor for an
-            // explanation on the structure of this object.
-            $scope.classification = {
-              answerGroupIndex: 0,
-              newOutcome: null
+            var fetchAndShowUnresolvedAnswers = function(expId, stateName) {
+              var unresolvedAnswersUrl = UrlInterpolationService.interpolateUrl(
+                '/createhandler/get_top_unresolved_answers/<exploration_id>', {
+                  exploration_id: expId
+                });
+              $http.get(unresolvedAnswersUrl, {
+                params: {
+                  state_name: _stateName
+                }
+              }).success(function(response) {
+                $scope.showUnresolvedAnswers(response.unresolved_answers);
+              }).error(function(response) {
+                $scope.showUnresolvedAnswers([]);
+              });
             };
 
-            FocusManagerService.setFocus('testInteractionInput');
+            $scope.showUnresolvedAnswers = function(unresolvedAnswers) {
+              $scope.showLoadingDots = false;
+              $scope.unresolvedAnswers = [];
+
+              unresolvedAnswers.forEach(function(item) {
+                var answer = item.answer;
+                var answerTemplate = (
+                  ExplorationHtmlFormatterService.getAnswerHtml(
+                    answer, stateInteractionIdService.savedMemento,
+                    stateCustomizationArgsService.savedMemento));
+                var classificationResult = (
+                  AnswerClassificationService.getMatchingClassificationResult(
+                    _explorationId, _stateName, _state, answer, rulesService));
+                var feedbackHtml = (
+                  classificationResult.outcome.feedback.getHtml());
+                var classificationType = (
+                  classificationResult.classificationCategorization);
+                if (classificationType !== EXPLICIT_CLASSIFICATION &&
+                  classificationType !== TRAINING_DATA_CLASSIFICATION &&
+                  !TrainingDataService.isConfirmedUnclassifiedAnswer(answer)) {
+                  $scope.unresolvedAnswers.push({
+                    answer: answer,
+                    answerTemplate: answerTemplate,
+                    classificationResult: classificationResult,
+                    feedbackHtml: feedbackHtml
+                  });
+                }
+              });
+            };
+
+            $scope.confirmToTrainingData = function(answerIndex) {
+              answer = $scope.unresolvedAnswers[answerIndex];
+              $scope.unresolvedAnswers.splice(answerIndex, 1);
+              if (answer.classificationResult.classificationCategorization ===
+                  DEFAULT_OUTCOME_CLASSIFICATION) {
+                TrainingDataService.trainDefaultResponse(answer.answer);
```

Comment:
It's not clear to me what trainDefaultResponse() means -- does it mean add the answer to the "default response" answer group? If so, maybe associateWIthDefaultResponse / associateWithAnswerGroup might be clearer?

(Also, normally I think we talk more about "training an ML model" rather than "training an answer group".)

Reply:
Yep. `trainDefaultResponse` adds the answer to confirmed unclassified answers. I'll modify the names.